src/main/java/org/mvplugins/multiverse/inventories/commands/InventoryModifyCommand.java

```diff
@@ -0,0 +1,146 @@
+import org.mvplugins.multiverse.external.jakarta.inject.Inject;
+import org.mvplugins.multiverse.external.jetbrains.annotations.NotNull;
+import org.mvplugins.multiverse.inventories.MultiverseInventories;
+import org.mvplugins.multiverse.inventories.profile.InventoryDataProvider;
+import org.mvplugins.multiverse.inventories.view.InventoryGUIHelper;
+import org.mvplugins.multiverse.inventories.view.ModifiableInventoryHolder;
+
+@Service
+final class InventoryModifyCommand extends InventoriesCommand {
+
+    private final InventoryDataProvider inventoryDataProvider;
+    private final MultiverseInventories inventories;
+    private final InventoryGUIHelper inventoryGUIHelper;
+
+    @Inject
+    InventoryModifyCommand(
+            @NotNull InventoryDataProvider inventoryDataProvider,
+            @NotNull MultiverseInventories inventories,
+            @NotNull InventoryGUIHelper inventoryGUIHelper
+    ) {
+        this.inventoryDataProvider = inventoryDataProvider;
+        this.inventories = inventories;
+        this.inventoryGUIHelper = inventoryGUIHelper;
+    }
+
+    // This method contains the logic for the /mvinv modify command
+    @Subcommand("modify")
+    @CommandPermission("multiverse.inventories.view.modify") // Specific permission for modification
+    @CommandCompletion("@mvinvplayernames @mvworlds")
+    @Syntax("<player> <world>")
+    @Description("Modify a player's inventory in a specific world.")
+    void onInventoryModifyCommand(
+            @NotNull MVCommandIssuer issuer,
+
+            // to make sure the command is only run by players
+            @Flags("resolve=issuerOnly")
+            @NotNull Player player,
+
+            @Syntax("<player>")
+            @Description("Online or offline player")
+            OfflinePlayer targetPlayer,
+
+            @Syntax("<world>")
+            @Description("The world the player's inventory is in")
+            MultiverseWorld world
+    ) {
+        if (player.getUniqueId().equals(targetPlayer.getUniqueId())) {
+            issuer.sendError(ChatColor.RED + "You cannot modify your own inventory using this command. Use your regular inventory.");
+            return;
+        }
+
+        String worldName = world.getName();
+        // Asynchronously load data using InventoryDataProvider
+        issuer.sendInfo(ChatColor.YELLOW + "Loading inventory data for " + targetPlayer.getName() + "...");
+
+        inventoryDataProvider.loadPlayerInventoryData(targetPlayer, worldName)
+                .thenAccept(playerInventoryData -> {
+                    // Ensure GUI operations run on the main thread
+                    Bukkit.getScheduler().runTask(inventories, () -> {
+                        // Create inventory with ModifiableInventoryHolder
+                        // Pass all necessary context to the holder for saving on close.
+                        Component title = Component.text("Modify " + targetPlayer.getName() + " @ " + worldName);
```

Comment:
Oh yea, don't use adventure text api, as we still have to maintain spigot compatibility for now.